services/sidecar/docker/entrypoint.sh

```diff
@@ -25,56 +30,78 @@ then
         if [ -z "$GROUPNAME" ]
         then
-            echo "$INFO" "mounted folder from $USERID, creating new group..."
-            GROUPNAME=host_group
-            addgroup -g "$GROUPID" $GROUPNAME
+            echo "$INFO" mounted folder from "$USERID", creating new group my"${SC_USER_NAME}"
+            GROUPNAME=my"${SC_USER_NAME}"
+            addgroup --gid "$GROUPID" "$GROUPNAME"
+            # change group property of files already around
+            find / -path /proc -prune -group "$SC_USER_ID" -exec chgrp --no-dereference "$GROUPNAME" {} \;
         else
-            echo "$INFO" "mounted folder from $USERID, adding scu to $GROUPNAME..."
-            addgroup scu $GROUPNAME
+            echo "$INFO" "mounted folder from $USERID, adding ${SC_USER_NAME} to $GROUPNAME..."
+            adduser "$SC_USER_NAME" "$GROUPNAME"
         fi
 
-        deluser scu > /dev/null 2>&1
-        adduser -u "$USERID" -G $GROUPNAME -D -s /bin/sh scu
+        echo "$INFO changing $SC_USER_NAME $SC_USER_ID:$SC_USER_ID to $USERID:$GROUPID"
+        deluser "${SC_USER_NAME}" > /dev/null 2>&1
+        if [ "$SC_USER_NAME" = "$GROUPNAME" ]
+        then
+            addgroup --gid "$GROUPID" "$GROUPNAME"
+        fi
+        adduser --disabled-password --gecos "" --uid "$USERID" --gid "$GROUPID" --shell /bin/sh "$SC_USER_NAME" --no-create-home
+        # change user property of files already around
+        find / -path /proc -prune -user "$SC_USER_ID" -exec chown --no-dereference "$SC_USER_NAME" {} \;
     fi
 fi
 
 
 
 # Appends docker group if socket is mounted
 DOCKER_MOUNT=/var/run/docker.sock
-
-
 if stat $DOCKER_MOUNT > /dev/null 2>&1
 then
-    GROUPID=$(stat -c %g $DOCKER_MOUNT)
+    echo "$INFO detected docker socket is mounted, adding user to group..."
+    GROUPID=$(stat --format=%g $DOCKER_MOUNT)
     GROUPNAME=scdocker
 
-
-    if ! addgroup -g "$GROUPID" $GROUPNAME > /dev/null 2>&1
+    if ! addgroup --gid "$GROUPID" $GROUPNAME > /dev/null 2>&1
     then
+        echo "$WARNING docker group with $GROUPID already exists, getting group name..."
         # if group already exists in container, then reuse name
-        GROUPNAME=$(getent group "${GROUPID}" | cut -d: -f1)
+        GROUPNAME=$(getent group "${GROUPID}" | cut --delimiters=: --fields=1)
+        echo "$WARNING docker group with $GROUPID has name $GROUPNAME"
     fi
-    addgroup scu "$GROUPNAME"
+    adduser "$SC_USER_NAME" "$GROUPNAME"
 fi
 
-echo "$INFO" "Starting boot ..."
-chown -R $USERNAME:"$GROUPNAME" /home/scu/input
-chown -R $USERNAME:"$GROUPNAME" /home/scu/output
-chown -R $USERNAME:"$GROUPNAME" /home/scu/log
+echo "$INFO ensuring write rights on folders ..."
+chown -R $USERNAME:"$GROUPNAME" "${SIDECAR_INPUT_FOLDER}"
+chown -R $USERNAME:"$GROUPNAME" "${SIDECAR_OUTPUT_FOLDER}"
+chown -R $USERNAME:"$GROUPNAME" "${SIDECAR_LOG_FOLDER}"
+
+echo "$INFO installing pythong dependencies..."
+cd services/sidecar || exit
+pip install --no-cache-dir -r requirements/dev.txt
```

Comment:
I guess this is temporary, right? If not, please check that we are in dev mode. I will check why it does not work in boot as non-root user and come back to you

Reply:
well yes temporary but for now I could not manage to make this work otherwise. also I am wondering if we always need both entrypoint and boot?

Reply:
I'll add the check good catch!